controllers/drplacementcontrol_controller.go

```diff
@@ -0,0 +1,1722 @@
+	vrgClientManifest := &mw.Spec.Workload.Manifests[0]
+	vrg := &rmn.VolumeReplicationGroup{}
+
+	err := yaml.Unmarshal(vrgClientManifest.RawExtension.Raw, &vrg)
+	if err != nil {
+		return false, errorswrapper.Wrap(err, fmt.Sprintf("unable to get vrg from ManifestWork %s", mw.Name))
+	}
+
+	return (vrg.Spec.ReplicationState == rmn.Primary), nil
+}
+
+func (d *DRPCInstance) ensureCleanup() error {
+	d.log.Info("ensure cleanup on secondaries")
+
+	if len(d.userPlacementRule.Status.Decisions) == 0 {
+		return nil
+	}
+
+	clusterToSkip := d.userPlacementRule.Status.Decisions[0].ClusterName
+
+	_, err := d.cleanup(clusterToSkip)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// cleanupPVForRestore cleans up required PV fields, to ensure restore succeeds to a new cluster, and
+// rebinding the PV to a newly created PVC with the same claimRef succeeds
+func (d *DRPCInstance) cleanupPVForRestore(pv *corev1.PersistentVolume) {
+	pv.ResourceVersion = ""
+	if pv.Spec.ClaimRef != nil {
+		pv.Spec.ClaimRef.UID = ""
+		pv.Spec.ClaimRef.ResourceVersion = ""
+		pv.Spec.ClaimRef.APIVersion = ""
+	}
+}
+
+func (d *DRPCInstance) ensureOldVRGMWOnClusterDeleted(clusterName string) (bool, error) {
+	d.log.Info("Ensuring that previous cluster VRG MW is deleted for ", "cluster", clusterName)
+
+	const done = true
+
+	mwName := BuildManifestWorkName(d.instance.Name, d.instance.Namespace, MWTypeVRG)
+	mw := &ocmworkv1.ManifestWork{}
+
+	err := d.reconciler.Get(d.ctx, types.NamespacedName{Name: mwName, Namespace: clusterName}, mw)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// the MW is gone.  Check the VRG if it is deleted
+			if d.ensureVRGDeleted(clusterName) {
+				return done, nil
+			}
+
+			return !done, nil
+		}
+
+		return !done, fmt.Errorf("failed to retrieve manifestwork %s from %s (%w)", mwName, clusterName, err)
+	}
+
+	d.log.Info("VRG ManifestWork exists", "name", mw.Name, "clusterName", clusterName)
+	// The VRG's MW exists. That most likely means that the VRG Object in the managed cluster exists as well. That also
+	// means that we either didn't attempt to set the VRG to secondary, or the MangedCluster may not be reachable.
+	// In either cases, we have to make sure that the VRG for the MW was set to secondary (if it is not, then set it) OR
+	// the MW has not been applied yet. In that later case, we will just need return DONE and we will wait for the
+	// MW status state to change to 'Applied' in order to proceed with deleting the MW, which causes the VRG to be deleted.
+	updated, err := d.hasVRGStateBeenUpdatedToSecondary(clusterName)
+	if err != nil {
+		return !done, fmt.Errorf("failed to check whether VRG replication state has been updated to secondary (%w)", err)
+	}
+
+	if !updated {
+		err = d.updateVRGStateToSecondary(clusterName)
+		// in either case, we need to wait for the MW to go to applied state
+		return !done, err
+	}
+
+	// if !IsManifestInAppliedState(mw) {
```

Comment:
This is removed because of the check failing to process "Available" conditions as well? IOW, is this temporary? (which is what I think it should be, but clarifying)

Reply:
That's why I just commented out the code rather than deleting it.  It is a temporary fix until we figure out how to handle that issue that you raised.